test/unitTests/core/test-issuers.js

```diff
@@ -0,0 +1,102 @@
+import { test } from 'tape-promise/tape';
+
+import { makeMint } from '../../../core/issuers';
+
+test('split and combine', t => {
+  try {
+    const mint = makeMint('fungible');
+    const issuer = mint.getIssuer();
+    const purse = mint.mint(1000);
+    const payments = [];
+    for (let i = 0; i < 100; i += 1) {
+      payments.push(purse.withdraw(1));
+    }
+
+    const combinedPayment = issuer.combine(payments);
+    t.deepEqual(combinedPayment.getBalance(), issuer.makeAmount(100));
+    for (const payment of payments) {
+      t.throws(() => payment.getBalance());
+    }
+
+    const badAmountsArray = Array(2).fill(issuer.makeAmount(10));
+    t.throws(_ => issuer.split(combinedPayment, badAmountsArray));
+
+    const goodAmountsArray = Array(10).fill(issuer.makeAmount(10));
+    const splitPayments = issuer.split(combinedPayment, goodAmountsArray);
+
+    for (const payment of splitPayments) {
+      t.deepEqual(payment.getBalance(), issuer.makeAmount(10));
+    }
+    t.throws(() => combinedPayment.getBalance());
+  } catch (e) {
+    t.assert(false, e);
+  } finally {
+    t.end();
+  }
+});
+
+test('depositExactly', async t => {
+  try {
+    const mint = makeMint('fungible');
+    const issuer = mint.getIssuer();
+    const purse = mint.mint(1000);
+    const targetPurse = issuer.makeEmptyPurse();
+    const payment = await purse.withdraw(7);
+    try {
+      await targetPurse.depositExactly(issuer.makeAmount(6), payment);
+      t.fail();
+    } catch (err) {
+      t.equal(err.message, 'payment balance must equal amount');
+    }
+    await targetPurse.depositExactly(7, payment);
```

Comment:
similarly, break this up here. good deposit and bad deposit can be separate test cases.

[and below.]

Reply:
Good idea